plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/Ck.hs

```diff
@@ -59,6 +59,8 @@ data CkValue uni fun =
   | VBuiltin (Term TyName Name uni fun ()) (BuiltinRuntime (CkValue uni fun))
     deriving (Show)
 
+instance ToBuiltinsRuntime DefaultFun (CkValue DefaultUni DefaultFun)
```

Comment:
I don't love the fact that now the functions exported from this module claim to work on any uni and fun, but you would need to add these instances to make it work in practice.

Reply:
Err, it should be a general instance in the CK machine, because we don't care about performance there. I originally did experiments in this module and then forgot to fix it. I'll do that.

The CEK machine is different though, there we do need the specialization. However note that `Cek.Internal` is not lying: it can handle absolutely any `uni` and `fun`, because it asks the caller to provide a `BuiltinsRuntime` for them and doesn't need anything else. The `Cek` module does expose general functions while only working on `DefaultUni` and `DefaultFun` out of the box though. Maybe we need a module that only exports specialized functions, so that there's no confusion, not sure.

Reply:
Err, not so fast, a general instance for conflicts with the one for `Either`...